byterun/domain.c

```diff
@@ -1055,7 +1054,9 @@ CAMLprim value caml_ml_domain_yield_until(value t)
     if (ts < caml_time_counter ()) {
       ret = Val_int(0); /* Domain.Sync.Timeout */
       break;
```

Comment:
My recollection is that this change was necessary to address an intermittently observed deadlock failure in the threads unit tests. Unfortunately, the test for _pr5325_ is still failing deterministically. Not sure why.  It does seem to be deadlocking on this same call to `caml_plat_timedwait`.